rpc/json/service.go

```diff
@@ -0,0 +1,200 @@
+package json
+
+import (
+	"errors"
+	"github.com/gorilla/rpc/v2/json2"
+	rpcclient "github.com/tendermint/tendermint/rpc/client"
+	"net/http"
+	"reflect"
+
+	ctypes "github.com/tendermint/tendermint/rpc/core/types"
+
+	"github.com/celestiaorg/optimint/rpc/client"
+)
+
+func GetHttpHandler(l *client.Client) (http.Handler, error) {
+	return newHandler(newService(l), json2.NewCodec()), nil
+}
+
+type method struct {
+	m        reflect.Value
+	argsType reflect.Type
+	returnType reflect.Type
+}
+
+
+func newMethod(m interface{}) *method {
+	mType := reflect.TypeOf(m)
+
+	return &method{
+		m:          reflect.ValueOf(m),
+		argsType:   mType.In(1).Elem(),
+		returnType: mType.Out(0).Elem(),
+	}
+}
+
+type service struct {
+	client  *client.Client
+	methods map[string]*method
+}
+
+func newService(c *client.Client) *service {
+	s := service{
+		client: c,
+	}
+	s.methods = map[string]*method{
+		"subscribe":            newMethod(s.Subscribe),
```

Comment:
Are these from somewhere? If not, can they be sorted alphabetically?

Reply:
It's all copy+paste+regexp from [tendermint](https://github.com/tendermint/tendermint/blob/master/internal/rpc/core/routes.go).